src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,205 @@
+                                                                           String.CASE_INSENSITIVE_ORDER);
+
+    Comparator<MatchmakerQueueMapPool> mapPoolComparator = Comparator.comparing(MatchmakerQueueMapPool::minRating, Comparator.nullsFirst(Double::compare))
+                                                                     .thenComparing(MatchmakerQueueMapPool::maxRating, Comparator.nullsLast(Double::compare));
+
+    SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedMap = new TreeMap<>(mapPoolComparator);
+
+    brackets.forEach((mapPool, mapVersions) -> {
+      List<MapVersion> sortedList = mapVersions.stream().sorted(mapVersionComparator).toList();
+      sortedMap.put(mapPool, sortedList);
+    });
+
+    return sortedMap;
+  }
+
+  private void removeDuplicates(SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets) {
+    Set<String> usedMaps = new HashSet<>();
+    sortedBrackets.replaceAll((k, v) -> sortedBrackets.get(k).stream().filter(mapVersion -> {
+      String name = mapVersion.map().displayName();
+      if (usedMaps.contains(name)) {
+        return false;
+      }
+      usedMaps.add(name);
+      return true;
+    }).collect(Collectors.toList()));
+  }
+
+  private Integer getPlayerBracketIndex(SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets, double rating) {
+    int i = 0;
+    for (Map.Entry<MatchmakerQueueMapPool, List<MapVersion>> entry : sortedBrackets.entrySet()) {
+      MatchmakerQueueMapPool pool = entry.getKey();
+      Double min = pool.minRating();
+      Double max = pool.maxRating();
+      if (min == null) {min = Double.NEGATIVE_INFINITY;}
+      if (max == null) {max = Double.POSITIVE_INFINITY;}
+
+      if (rating < max && rating > min) {
+        return i;
+      }
+      i++;
+    }
+    return null;
+  }
+
+  private void addMapTile(MapVersion mapVersion) {
+    int i = 0;
+    List<Integer> indexes = new ArrayList<>();
+    for (List<MapVersion> maps : this.sortedBracketsWithDuplicates.values()) {
+      if (maps.contains(mapVersion)) {
+        indexes.add(i);
+      }
+      i++;
+    }
+    int diff = Collections.min(indexes.stream().map(idx -> Math.abs(idx - this.playerBracketIndex)).toList());
+    double relevanceLevel = switch (diff) {
+      case 0 -> 1;
+      case 1 -> 0.2;
+      default -> 0;
+    };
+
+    TeamMatchmakingMapTileController controller = uiService.loadFxml(
+        "theme/play/teammatchmaking/matchmaking_map_tile.fxml");
+    controller.setRelevanceLevel(relevanceLevel);
+    controller.setMapVersion(mapVersion);
+    this.tilesContainer.getChildren().add(controller.getRoot());
+  }
+
+  private void resizeToContent(int tilecount, int tileSize) {
+    double maxWidth = (this.maxWidth != 0) ? this.maxWidth :  Screen.getPrimary().getVisualBounds().getWidth();
+    double maxHeight = (this.maxHeight != 0) ? this.maxHeight : Screen.getPrimary().getVisualBounds().getHeight();
```

Comment:
Making these properties will remove the need to use the Screen static methods which could cause issues in tests.